docker/usr/local/toolchain/generate_toolchains.py

```diff
@@ -107,11 +107,11 @@ def generate_toolchain_script(target):
         export PKG_CONFIG_PATH="/usr/local/lib/pkgconfig"
 
         # Set our optimisation flags
-        export CFLAGS="{flags}"
-        export CXXFLAGS="${{CFLAGS}}"
-        export CPPFLAGS="${{CFLAGS}}"
-        export FFLAGS="${{CFLAGS}}"
-        export FCFLAGS="${{CFLAGS}}"
+        export CFLAGS="${{CFLAGS}} {flags}"
+        export CXXFLAGS="${{CXXFLAG}} ${{CFLAGS}}"
+        export CPPFLAGS="${{CPPFLAGS}} ${{CFLAGS}}"
+        export FFLAGS="${{FFLAGS}} ${{CFLAGS}}"
+        export FCFLAGS="${{FCFLAGS}} ${{CFLAGS}}"
```

Comment:
So this just forwards the environment along with everything else? Does this only work for autotools or cmake etc as well

Reply:
My understanding that cmake uses `CMAKE_<LANG>_FLAGS_INIT` and `<LANG>FLAGS` to initialise `CMAKE_<LANG>_FLAGS`. So I think it combines both of those and so it should.

We have no need to test any of the other build tools (cmake, meson, boost), so we wont know until we need to.